src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -337,6 +337,82 @@ .maxstack  7
         }
 
         [Fact]
+        public void SimpleTupleNew2()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        var x = new (int a, int b)(1, 2);
+        System.Console.WriteLine(x.a.ToString());
+    }
+}
+" + trivial2uple;
+
+            var comp = CompileAndVerify(source, parseOptions: TestOptions.Regular.WithTuplesFeature(), expectedOutput: @"
+1
+");
+            comp.VerifyDiagnostics();
+            comp.VerifyIL("C.Main", @"
+{
+  // Code size       27 (0x1b)
+  .maxstack  3
+  .locals init (System.ValueTuple<int, int> V_0) //x
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  ldc.i4.1
+  IL_0003:  ldc.i4.2
+  IL_0004:  call       ""System.ValueTuple<int, int>..ctor(int, int)""
+  IL_0009:  ldloca.s   V_0
+  IL_000b:  ldflda     ""int System.ValueTuple<int, int>.Item1""
+  IL_0010:  call       ""string int.ToString()""
+  IL_0015:  call       ""void System.Console.WriteLine(string)""
+  IL_001a:  ret
+}");
+        }
+
+        [WorkItem(10874, "https://github.com/dotnet/roslyn/issues/10874")]
+        [Fact]
+        public void SimpleTupleNew3()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        var x0 = new (int a, int b)(1, 2, 3);
+        System.Console.WriteLine(x0.ToString());
+
+        var x1 = new (int, int)(1, 2, 3);
+        System.Console.WriteLine(x1.ToString());
+
+        var x2 = new (int, int)(1, ""2"");
+        System.Console.WriteLine(x2.ToString());
+
+        var x3 = new (int, int)(1);
+        System.Console.WriteLine(x3.ToString());
+    }
+}
+" + trivial2uple;
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular.WithTuplesFeature());
+            comp.VerifyDiagnostics(
+                // (6,22): error CS1729: 'ValueTuple<int, int>' does not contain a constructor that takes 3 arguments
```

Comment:
Possibly. 
Technically ctor belongs to the underlying type, so the error is correct - it tells exactly what type is missing a constructor. 

I can see, however, how a custom error "this tuple cannot be constructed with N elements" could be more helpful.

Captured in https://github.com/dotnet/roslyn/issues/10874